designdocs/permissions.md

```diff
@@ -0,0 +1,87 @@
+# proposal for permissions request
+
+
+### Philosophy
+The user-agent will always be closer to the user than a spec. The user agent has greater access to current conditions than we (the spec writers) ever will. As a philosophy we should give the user agent *as much freedom as possible to “do the right thing”* under whatever circumstances are at the time a web application is started.
+
+### scope
+
+Currently we are having difficulties finding a solution that all parties can agree to. This is about the permissions for various properties in WebXR.  This proposal addresses the following concerns
+
+* what permissions are available to be asked for
+* how granular are these permissions
+* are the permissions granted at page load / application start
+* or are the permissions granted at the time of use.
+
+### stating concerns
+
+Concerns about granting at page load time are
+* users may be requested to approve multiple permissions one after another, creating click fatigue.
+* if all apps request permissions at start, the user has no information to judge whether or not they should grant this, because they haven’t entered the experience yet.
+
+Concerns about granting requests once already inside of immersive mode are:
+* access requests when inside immersive mode should be done with some sort of a secure method, such as a dedicated hardware button, so that applications cannot spoof such permissions.
+* some devices or scenarios may not be able to provide such a secure method, thus doing the request before entering immersive mode is preferable.
+
+Concerns about both systems include:
+* the user agents know more about the user, the current scenario, and can make better decisions *at the time of use* than a spec can. Neither method give the user agent freedom to adapt.
+* there may be new attacks devised or new flaws discovered after we have shipped, but user agents will not be able to address these because they are constrained by the spec.
+
+There are most likely more concerns than just what I have listed above.
+
```

Comment:
Another goal is that developers should generally know what to expect across user agents (i.e., what may cause a prompt and when) and that the user experience should be relatively consistent across user agents.

Reply:
Is there agreement on this point?  From what I can tell, some UAs want to do just-in-time permissions but others want them upfront.  As long as a single code path exists that will hit both, is there a specific reason UAs should be forced to conform?

Reply:
The point of this proposal is to give UAs freedom while giving developers only a single path to code.

Reply:
I think the specification needs to make it clear to developers when there might be consent prompts so they can avoid patterns that may cause undesirable results. It would also help with compatibility if there are fewer ways things can vary across implementations.